File: Data Structures & Algorithms/reverse-integer/submission-0.py
class Solution:
    def reverse(self, x: int) -> int:
        tmp=x
        negative=False
   
        if tmp<0:
            negative=True
            tmp*=-1
        start=0
        while tmp>0:
            start=start*10 + (tmp%10)
            tmp=tmp//10
        if negative:
            start*=-1
        if start<= -2**31 or start>=2**31 -1:
            return 0    
        return start

File: Data Structures & Algorithms/merge-intervals/submission-0.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,(a,b)->Integer.compare(a[0],b[0]));
        List<int[]> res = new ArrayList<>();
        res.add(intervals[0]);
        for(int [] i: intervals){
            int start=i[0];
            int end=i[1];
            int previousEnd=res.get(res.size()-1)[1];
            if(start<=previousEnd){
                res.get(res.size()-1)[1]=Math.max(previousEnd,end);
            }else{
                res.add(new int[]{start,end});
            }
        }
        return res.toArray(new int[res.size()][]);
    }
}

//[[1,3],[1,5],[1,7]], [1,7]
